### ISSUE TITLE
Requirements for Cutils branch to be merged into master

### ISSUE THREAD

**Issue by Lemonymous** [kartoFlane/ITB-ModLoader#33]
I am uncertain of what needs to be done before the Cutils branch can be merged into master.
In an effort to let more people help out with writing tests and other required code, would it be possible to create some sort of checklist and/or a guide for the required specifications for this to happen?

**Comment by kartoFlane** [comment]
Before merging, I wanted to have test coverage for as many of the newly-added Cutils functions as possible. 

There are also several functions that are present in lua, but apparently don't work (GetWeaponName, GetRepairSkill?), or are yet to be implemented (there are a few TODOs left over in pawn and board files). These will have to be removed or added, as appropriate.

Ideally, tests would verify the newly-added features functionally, ie. check that they produce the expected results without relying on their associated setter/getter for verification. For example, the test for `SetMaxHealth` should use the function to set the pawn's max health, but heal it back up using non-Cutil mechanism, damage it for its old max health; expect that the pawn did not die, and chech that the leftover health is equal to the difference between old and new max health values (currently it doesn't do that, I'll have to fix that).

For some functions this may not be feasible. In those cases using the associated setter/getter is fine, as long as the other function can be tested independently, as described above.

Some other tests (particularly GameState) might require implementing a completely different approach, eg. scheduled to run when the mod loader is in development mode and a condition is met (to test transitions and such).

In the worst case that a function can't be tested *at all*, it should at least be invoked to verify that it doesn't crash the game.

In checklist form, it'd look like this (checked boxes mean that the test is already implemented):

- [ ] Pawns:
  - [ ] GetUndoLoc
  - [ ] SetUndoLoc
  - [x] GetOwner
  - [x] SetOwner
  - [x] SetFire
  - [ ] IsHighlighted (probably not testable)
  - [x] GetImpactMaterial
  - [x] SetImpactMaterial
  - [x] GetColor
  - [x] SetColor
  - [ ] IsPlayerControlled
  - [x] IsMassive
  - [x] SetMassive
  - [x] IsMovementAvailable
  - [x] SetMovementAvailable
  - [x] SetFlying
  - [ ] IsTeleporter
  - [ ] SetTeleporter
  - [ ] IsJumper
  - [ ] SetJumper
  - [ ] GetMaxHealth
  - [ ] GetBaseMaxHealth
  - [ ] SetHealth
  - [x] SetMaxHealth
  - [ ] SetBaseMaxHealth
  - [ ] MarkHpLoss (probably not testable)
  - [x] GetWeaponCount
  - [x] GetWeaponType
  - [x] GetWeaponClass
  - [ ] GetMoveSkill
  - [ ] SetMoveSkill
  - [ ] RemoveWeapon
  - [ ] GetPilot
  - [ ] SetMech
  - [ ] SetFrozen
  - [ ] SetShield
  - [ ] SetAcid
  - [ ] AddWeapon
- [ ] Board:
  - [ ] SetFire
  - [ ] IsForest
  - [ ] IsForestFire
  - [ ] IsShield
  - [ ] SetShield
  - [ ] SetWater
  - [ ] SetAcidWater
  - [ ] GetHealth
  - [ ] GetMaxHealth
  - [ ] GetLostHealth
  - [ ] SetHealth
  - [ ] SetMaxHealth
  - [ ] SetBuilding
  - [ ] SetMountain
  - [ ] SetIce
  - [ ] SetRubble
  - [ ] SetSnow
  - [ ] SetUniqueBuilding
  - [ ] RemoveUniqueBuilding
  - [ ] RemoveItem
  - [ ] GetItemName
  - [ ] GetHighlighted (proably not testable)
  - [ ] IsHighlighted (probably not testable)
  - [ ] MarkGridLoss (probably not testable)
  - [ ] IsGameBoard
  - [ ] SetTerrain
  - [ ] IsTerrain
  - [ ] SetFrozen
- [ ] Game
  - [ ] GetRep
  - [ ] GetPow
  - [ ] GetMaxPow
  - [ ] GetCores
  - [ ] GetGridDef
  - [ ] AddRep
  - [ ] AddPow
  - [ ] AddMaxPow
  - [ ] AddCores
  - [ ] AddGridDef
  - [ ] IsCogHovered
  - [ ] IsRepHovered
  - [ ] IsCoreHovered
  - [ ] IsPowerHovered
  - [ ] IsGridDefHovered
  - [ ] IsPeopleHovered
- [ ] GameState
  - [ ] IsMainMenu
  - [ ] IsHangar
  - [ ] IsRegion
  - [ ] IsMission
  - [ ] IsTransition
  - [ ] IsEnteringHangar
  - [ ] IsLeavingHangar
- [ ] Region
  - [ ] GetSelectedMech
  - [ ] GetHoveredMech
- [ ] MechEdit
  - [ ] GetItemOnCursor
  - [ ] IsTipImage
  - [ ] IsTestHighlightable
  - [ ] IsClassHighlightable
  - [ ] IsNameHighlightable
  - [ ] IsInstallHighlightable
  - [ ] IsUndoHighlightable
  - [ ] IsHealthHighlightable
  - [ ] IsHealthIconHighlightable
  - [ ] IsMoveHighlightable
  - [ ] IsMoveIconHighlightable
  - [ ] IsInventoryHighlightable
  - [ ] IsInventoryUpHighlightable
  - [ ] IsInventoryDownHighlightable
  - [ ] IsSkill1Highlightable
  - [ ] IsSkill2Highlightable
  - [ ] IsAbilityHighlightable
  - [ ] IsTestHovered
  - [ ] IsClassHovered
  - [ ] IsNameHovered
  - [ ] IsInstallHovered
  - [ ] IsReactorHovered
  - [ ] IsUndoHovered
  - [ ] IsHealthHovered
  - [ ] IsHealthIconHovered
  - [ ] IsMoveHovered
  - [ ] IsMoveIconHovered
  - [ ] IsInventoryHovered
  - [ ] IsInventoryUpHovered
  - [ ] IsInventoryDownHovered
  - [ ] IsSkill1Hovered
  - [ ] IsSkill2Hovered
  - [ ] IsAbilityHovered
  - [ ] SetTestHighlightable
  - [ ] SetClassHighlightable
  - [ ] SetNameHighlightable
  - [ ] SetPilotNameHighlightable
  - [ ] SetInstallHighlightable
  - [ ] SetHealthIconHighlightable
  - [ ] SetMoveHighlightable
  - [ ] SetMoveIconHighlightable
